### PR TITLE
Adds a message telling you to npm install when there's missing deps

### PR DESCRIPTION
**Pull request by orta**
Should make the next step very obvious if you hit https://github.com/DefinitelyTyped/DefinitelyTyped/issues/24788

### CODE REVIEW

**Comment by orta** [comment]
Looks like:

```
/Users/ortatherox/dev/typescript/DefinitelyTyped/types/gatsby-plugin-breakpoints
Error: A module look-up failed, the most common case for this error a dependencies which needs to be `npm installed`.
        
Before you debug, first try running: 
   npm install --prefix types/react

Then re-run. Full error logs are below.

Errors in typescript@4.3 for external dependencies:
../react/index.d.ts(38,22): error TS2307: Cannot find module 'csstype' or its corresponding type declarations.

    at /Users/ortatherox/dev/typescript/dtslint/bin/index.js:207:19
    at Generator.next (<anonymous>)
    at fulfilled (/Users/ortatherox/dev/typescript/dtslint/bin/index.js:6:58)
```

File: src/lint.ts [hunk 1]
```diff
@@ -76,7 +76,26 @@ function testDependencies(
...
+    // Add an edge-case for someone needing to `npm install` in react when they first edit a DT module which depends on it - #226
+    const cannotFindDepsDiags = diagnostics.find(d => d.code === 2307 && d.messageText.toString().includes("Cannot find module"))
+    if (cannotFindDepsDiags && cannotFindDepsDiags.file) {
+        const path = cannotFindDepsDiags.file.fileName
+        const typesFolder = basename(dirname(path))
+
+        return `A module look-up failed, the most common case for this error are dependencies which needs to be \`npm installed\` before you can lint.
+        
+Before you debug, first try running: 
+   npm install --prefix types/${typesFolder}
```
Comment (sandersn) [inline src/lint.ts:91]: does this work if you run the `npm test` command from inside `types/` or `types/whatever/`?

**Review by sandersn — APPROVED** [review]
I like this workaround!

File: src/lint.ts (hunk 1 above)
Comment (orta) [reply src/lint.ts]: Good idea, it's now absolute so it doesn't matter

**Comment by orta** [comment]
Gonna drop travis in an upcoming PR, GH Actions shows green.

**Pull request merged**